nilearn/surface/surface.py

```diff
@@ -968,160 +964,12 @@ def _gifti_img_to_mesh(gifti_img):
-                "it must have two elements, the first is "
-                "a Numpy array containing the x-y-z coordinates "
-                "of the mesh vertices, the second is a Numpy "
-                "array containing  the indices (into coords) of "
-                "the mesh faces. The input was a list with "
-                f"{len(surf_mesh)} elements."
-            )
-    elif hasattr(surf_mesh, "faces") and hasattr(surf_mesh, "coordinates"):
-        coords, faces = surf_mesh.coordinates, surf_mesh.faces
-        mesh = Mesh(coordinates=coords, faces=faces)
-
-    else:
-        raise ValueError(
-            "The input type is not recognized. "
-            "Valid inputs are one of the following file "
-            "formats: .gii, .gii.gz, "
-            "Freesurfer specific files such as "
-            f"{_stringify(FREESURFER_MESH_EXTENSIONS)}"
-            "or two Numpy arrays organized in a list, tuple or "
-            'a namedtuple with the fields "coordinates" and '
-            '"faces"'
-        )
-
-    return mesh
-
-
-def load_surface(surface):
-    """Load a surface.
-
-    Parameters
-    ----------
-    surface : Surface-like (see description)
-        The surface to be loaded.
-        A surface can be:
-            - a nilearn.surface.Surface
-            - a sequence (mesh, data) where:
-                - :term:`mesh` can be:
-                    - a nilearn.surface.Mesh
-                    - a path to .gii or .gii.gz etc.
-                    - a sequence of two numpy arrays,
-                    the first containing :term:`vertex` coordinates
-                    and the second containing triangles.
-                - data can be:
-                    - a path to .gii or .gii.gz etc.
-                    - a numpy array with shape (n_vertices,)
-                    or (n_time_points, n_vertices)
-
-    Returns
-    -------
-    surface : Surface
-        With the fields "mesh" (Mesh object) and "data" (:obj:`numpy.ndarray`).
-
-    """
-    # Handle the case where we received a Surface
-    # object with mesh and data attributes
-    if hasattr(surface, "mesh") and hasattr(surface, "data"):
-        mesh = load_surf_mesh(surface.mesh)
-        data = load_surf_data(surface.data)
-    # Handle the case where we received a sequence
-    # (mesh, data)
-    elif isinstance(surface, (list, tuple, np.ndarray)):
-        if len(surface) != 2:
-            raise ValueError(
-                "`load_surface` accepts iterables "
-                "of length 2 to define a surface. "
-                f"You provided a {type(surface)} "
-                f"of length {len(surface)}."
-            )
-        mesh = load_surf_mesh(surface[0])
-        data = load_surf_data(surface[1])
-    else:
-        raise ValueError(
-            "Wrong parameter `surface` in `load_surface`. "
-            "Please refer to the documentation for more information."
-        )
-    return Surface(mesh, data)
-
-
 def check_mesh(mesh):
```

Comment:
Ok so this function is only used in `plot_img_on_surf` to ensure that we only plot Nifti stat maps on a FreeSurfer-like mesh.

I think it could be named better. Something like `check_mesh_is_fsaverage`. I can add a deprecation warning saying that it will be renamed. 

Or we skip deprecating this one because as @Remi-Gau pointed out here: https://github.com/nilearn/nilearn/issues/4789#issuecomment-2497914202 it wasn't shown in the public docs.

Reply:
no deprecation needed: not in the doc and not in the [surface/__init__.py](https://github.com/nilearn/nilearn/blob/main/nilearn/surface/__init__.py)